File: java64/nivel4/exer42.java
import javax.swing.JOptionPane;

public class exer42 {
    public static void main(String[] args) {
        int valor, menor, maior;

        valor = Integer.parseInt(JOptionPane.showInputDialog("Informe o valor: "));
        menor = valor;
        maior = valor;

        do {
            valor = Integer.parseInt(JOptionPane.showInputDialog("Informe o valor: "));

            if (valor > maior) {
                maior = valor;
            }

            if (valor < menor) {
                menor = valor;
            }
        } while (valor != 0);
        JOptionPane.showMessageDialog(null, "Maior: " + maior + "\nMenor: " + menor);
    }
}

File: pyhton64/nivel4/exer42.py
print('0 para')
valor = int(input('Valor: '))
maior = menor = valor
if valor != 0:
    while valor != 0:
        valor = int(input('Valor: '))

        if valor > maior:
            maior = valor
        elif valor < menor:
            menor = valor
    print(f'Maior: {maior}\nMenor: {menor}')
else:
    print('Saida')
